### PR TITLE
feat(middleware/proxy): Add DialDualStack option for upstream IPv6 support

### LINKED ISSUE #2890: 🐛 [Bug]: Proxy middleware and Vite dev server on localhost, tcp4 127.0.0.1:5173: connect: connection refused
### Bug Description

When using the proxy middleware to `proxy.BalanceForward([]string{"http://localhost:5173"}))` to a Vite/react dev server running on port 5173 the connection is refused with the following output `Error: The HTTP request failed with error dial tcp4 127.0.0.1:5173: connect: connection refused`

### How to Reproduce

Clone https://github.com/sixcolors/FiberReactTest

```bash
cd frontend
npm install --include=dev && npm run dev
```

The proxy demonstrates this behaviour as can be seen in the main.go:

```go
if DevMode {
		app.Get("*", proxy.BalancerForward([]string{"http://localhost:5173"}))
```

However the issue has been isolated to the fasthttp client.

To verify run the following:

```go
package main

import (
	"io"
	"log"
	"net/http"

	"github.com/valyala/fasthttp"
)

func main() {
	// test connection to http://localhost:5173
	resp, err := http.Get("http://localhost:5173")
	if err != nil {
		log.Fatalf("Error: The HTTP request failed with error %s", err)
	} else {
		defer resp.Body.Close()
		data, err := io.ReadAll(resp.Body)
		if err != nil {
			log.Fatalf("Error reading body: %v", err)
		}
		log.Printf("HTTP Response Status: %d %s", resp.StatusCode, http.StatusText(resp.StatusCode))
		log.Printf("Response Body: %s", string(data))
	}

	statusCode, body, err := fasthttp.Get(nil, "http://localhost:5173")
	if err != nil {
		log.Fatalf("Error: The HTTP request failed with error %s", err)
	} else {
		log.Printf("HTTP Response Status: %d", statusCode)
		log.Printf("Response Body: %s", body)
	}
}
```

### Expected Behavior

fasthttp client should connect to http://localhost:5173 and the proxy should work.

### Actual Behavior 

fasthttp client will not try dialing IPv6 by default and only IPv4 is tried, resulting in a connection refused error. 

### Fiber Version

v2.52.2

### Code Snippet (optional)

_No response_

### Checklist:

- [X] I agree to follow Fiber's [Code of Conduct](https://github.com/gofiber/fiber/blob/master/.github/CODE_OF_CONDUCT.md).
- [X] I have checked for existing issues that describe my problem prior to opening this one.
- [X] I understand that improperly formatted bug reports may be closed without explanation.

### PR DESCRIPTION
**Pull request by negrel**
## Description

This PR adds a `DialDualStack` options to `github.com/gofiber/fiber/v2/middleware/proxy.Config` in order to enable proxy middleware to forward to IPv6 upstreams.

Fixes #2890

## Changes introduced

List the new features or adjustments introduced in this pull request. Provide details on benchmarks, documentation updates, changelog entries, and if applicable, the migration guide.

- [ ] Benchmarks: Describe any performance benchmarks and improvements related to the changes.
- [x] Documentation Update: Detail the updates made to the documentation and links to the changed files.
- [ ] Changelog/What's New: Include a summary of the additions for the upcoming release notes.
- [ ] Migration Guide: If necessary, provide a guide or steps for users to migrate their existing code to accommodate these changes.
- [ ] API Alignment with Express: Explain how the changes align with the Express API.
- [ ] API Longevity: Discuss the steps taken to ensure that the new or updated APIs are consistent and not prone to breaking changes.
- [ ] Examples: Provide examples demonstrating the new features or changes in action.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [ ] Enhancement (improvement to existing features and functionality)
- [ ] Documentation update (changes to documentation)
- [ ] Performance improvement (non-breaking change which improves efficiency)
- [ ] Code consistency (non-breaking change which improves code reliability and robustness)

## Checklist

Before you submit your pull request, please make sure you meet these requirements:

- [ ] Followed the inspiration of the Express.js framework for new functionalities, making them similar in usage.
- [x] Conducted a self-review of the code and provided comments for complex or critical parts.
- [x] Updated the documentation in the `/docs/` directory for [Fiber's documentation](https://docs.gofiber.io/).
- [x] Added or updated unit tests to validate the effectiveness of the changes or new features.
- [x] Ensured that new and existing unit tests pass locally with the changes.
- [x] Verified that any new dependencies are essential and have been agreed upon by the maintainers/community.
- [x] Aimed for optimal performance with minimal allocations in the new code.
- [ ] Provided benchmarks for the new code to analyze and improve upon.

## Commit formatting

Please use emojis in commit messages for an easy way to identify the purpose or intention of a commit. Check out the emoji cheatsheet here: [CONTRIBUTING.md](https://github.com/gofiber/fiber/blob/master/.github/CONTRIBUTING.md#pull-requests-or-commits)

## Summary by CodeRabbit

- **New Features**
	- Introduced a new configuration option `DialDualStack` for proxy settings, allowing clients to attempt connections to both IPv4 and IPv6 host addresses.
- **Documentation**
	- Updated API documentation to include the `DialDualStack` configuration option.
- **Tests**
	- Added new test functions for IPv4 and IPv6 network configurations in proxy settings.

### CODE REVIEW

File: docs/api/middleware/proxy.md [hunk 1]
```diff
@@ -151,6 +151,7 @@ app.Use(proxy.BalancerForward([]string{
 | ReadBufferSize  | `int`                                          | Per-connection buffer size for requests' reading. This also limits the maximum header size. Increase this buffer if your clients send multi-KB RequestURIs and/or multi-KB headers (for example, BIG cookies). | (Not specified) |
 | WriteBufferSize | `int`                                          | Per-connection buffer size for responses' writing.                                                                                                                                                             | (Not specified) |
 | TlsConfig       | `*tls.Config` (or `*fasthttp.TLSConfig` in v3) | TLS config for the HTTP client.                                                                                                                                                                                | `nil`           |
+| DialDualStack       | `bool`                                     | Client will attempt to connect to both ipv4 and ipv6 host addresses if set to true.                                                                                                                            | `false`           |
```
[the remark this answers is not in the document]
Comment (ReneWerner87) [reply docs/api/middleware/proxy.md:154]: @negrel 
> Consider adding a practical example or a more detailed explanation of the DialDualStack option's impact on proxy behavior to aid user understanding.

is a good hint, could you do this

only a feature that is understandable and well documented will be found and used

[the remark this answers is not in the document]
Comment (negrel) [reply docs/api/middleware/proxy.md]: @ReneWerner87 I've added an example. Let me know if that suits you!

**Comment by gaby** [comment]
@negrel Can you take a look at the merge conflict so we can move this forward? Thanks!

**Comment by negrel** [comment]
@gaby I've fix all conflicts. Let me know if this is good for you!

**Comment by gaby** [comment]
> @gaby I've fix all conflicts. Let me know if this is good for you!

Running CI now :-) thanks!

**Review by gaby — APPROVED** [review]
LGTM

**Review by sixcolors — APPROVED** [review]
@negrel tested with vite dev server fixes: #2890. LGTM, thanks for adding this.

**Pull request merged**